cumulus_library/studies/core/builder_core_medication.py

```diff
@@ -0,0 +1,95 @@
+    display_text = "Creating core medication table..."
+
+    def _check_data_in_fields(self, data_types: dict, cursor, schema: str):
+        """Validates whether either observed medication source is present"""
+        with get_progress_bar(transient=True) as progress:
+            task = progress.add_task(
+                "Detecting available medication sources...",
+                total=5,
+            )
+
+            # Validating medication requests
+            query = get_is_table_not_empty_query(
+                "medicationrequest", "medicationreference"
+            )
+            cursor.execute(query)
+            progress.advance(task)
+            if cursor.fetchone() is None:
+                return data_types
+            query = get_column_datatype_query(
+                schema, "medicationrequest", "medicationreference"
+            )
+            cursor.execute(query)
+            progress.advance(task)
+            if "reference" not in cursor.fetchone()[0]:
+                return data_types
+            query = get_is_table_not_empty_query(
+                "medicationrequest", "medicationreference.reference"
+            )
+            cursor.execute(query)
+            progress.advance(task)
+            if cursor.fetchone() is None:
+                return data_types
+
+            # checking med ref contents for our two linkage cases
+            query = get_is_table_not_empty_query(
+                "medicationrequest",
+                "medicationreference.reference",
+                conditions=["medicationreference.reference LIKE '#%'"],
+            )
+            cursor.execute(query)
+            progress.advance(task)
+            if cursor.fetchone() is not None:
+                data_types["by_contained_ref"] = True
+            query = get_is_table_not_empty_query(
+                "medicationrequest",
+                "medicationreference.reference",
+                conditions=["medicationreference.reference LIKE 'Medication/%'"],
+            )
+            cursor.execute(query)
+            progress.advance(task)
+            if cursor.fetchone() is not None:
+                data_types["by_external_ref"] = True
+            return data_types
+
+    def prepare_queries(self, cursor: object, schema: str) -> dict:
+        """Constructs queries related to condition codeableConcept
+
+        :param cursor: A database cursor object
+        :param schema: the schema/db name, matching the cursor
+
+        """
+        medication_datasources = {
+            "by_contained_ref": False,
+            "by_external_ref": False,
+        }
+        medication_datasources = self._check_data_in_fields(
+            medication_datasources, cursor, schema
+        )
+        if (
+            medication_datasources["by_contained_ref"]
+            or medication_datasources["by_external_ref"]
+        ):
+            self.queries.append(get_core_medication_query(medication_datasources))
+        else:
```

Comment:
So i'm not sure what you mean here - if either, or both, of these data sources are present, we call the templater and it will spit out either the singleton case or the union case. if neither is present, we create an empty table so joins won't fail.

Reply:
Ah OK I think I just misunderstood the `else:` branch - I thought the else branch was "OK don't worry about these cases, just look at inline codes". So this PR is still just about unioning the two reference cases - at present, we're still leaving the union of all three cases (inline, external ref, contained ref) as an exercise for the reader/study-author. That's fine, I just misunderstood this code.

Reply:
oh ok wait - i think i get it now

Reply:
for posterity - we are going to add in the inline case in an immediate followon, it just never ended up in the ticket i was working from so it got missed.